packages/gatsby-plugin-sharp/src/utils.js

```diff
@@ -88,33 +88,48 @@ export function rgbToHex(red, green, blue) {
     .slice(1)}`
 }
 
+const DEFAULT_FLUID_SIZE = 800
 export const calculateImageSizes = ({
   file,
+  imgDimensions,
   width,
   maxWidth,
   height,
   maxHeight,
   layout,
-  outputPixelDensities,
+  outputPixelDensities = [0.25, 0.5, 1, 2, 3],
   srcSetBreakpoints,
 }) => {
+  if (maxWidth === 0 || width === 0 || maxHeight === 0 || height === 0) {
```

Comment:
Check that they're positive too

Reply:
Just added some better checking on this in https://github.com/gatsbyjs/gatsby/pull/27554/commits/f90f4bfb5cb0377751694384234d2ed19c6477e5